torch/csrc/jit/backends/backend_init.cpp

```diff
@@ -0,0 +1,126 @@
+
+          // Clone orig_module to make sure backend transformation is
+          // functional.
+          auto cloned_module = orig_module.clone();
+
+          // Represents of a Type of Dict[str, Any].
+          auto any_dict_ty =
+              DictType::create(StringType::get(), AnyType::get());
+
+          // Call preprocess.
+          Stack stack;
+          push(stack, cloned_module._ivalue());
+          push(stack, toIValue(extra_infos, any_dict_ty));
+          backend.preprocess(stack);
+          IValue preprocessed_module = pop(stack);
+
+          // Generate LoweredModule.
+          Module loweredModule("torch.jit." + backend.name + "LoweredModule");
+
+          // Generate attributes.
+          // This is for the extra_infos passed in to to_<backend> or loaded
+          // from an export model.
+          loweredModule.register_attribute(
+              "__extra_infos", any_dict_ty, toIValue(extra_infos, any_dict_ty));
+
+          // This is the list of opaque backend handles returned by
+          // backend.compile.
+          loweredModule.register_attribute(
+              "__backend_handles",
+              any_dict_ty,
+              c10::impl::GenericDict(
+                  any_dict_ty->getKeyType(), any_dict_ty->getValueType()));
+
+          // This is the original cloned and preprocessed module.
+          loweredModule.register_attribute(
+              "__orig_module", AnyType::get(), preprocessed_module);
+
+          // Methods.
+          loweredModule.define(R"(
+            def __getstate__(self):
+                return self.__extra_infos, self.__orig_module
+            )");
+
+          // This is a convenient wrapper for backend.compile.
+          loweredModule.define(
+              "\ndef __compile__(self):\n"
+              //"\tself.__backend_handles.clear()"
+              "\thandles: List[Any] = torch.ops." +
+              backend.name +
+              ".compile(self, self.__orig_module, self.__extra_infos)\n"
+              "\tfor h, k in zip(handles, self.__extra_infos.keys()):\n"
+              "\t\tself.__backend_handles[k] = h\n");
+
+          loweredModule.define(R"(
+            def __setstate__(self, state):
+                self.__extra_infos = state[0]
+                self.__orig_module = state[1]
+                self.__compile__()
+            )");
+
+          // This loop generates one method on the LoweredModule for every key
+          // in extra_infos.
+          for (auto& e : extra_infos) {
+            std::string method_name = py::cast<std::string>(e.first);
+
+            loweredModule.define(
+                "\ndef " + method_name + "(self, input):\n\treturn torch.ops." +
+                backend.name + ".execute(self, self.__backend_handles[\"" +
+                method_name + "\"], input)\n");
+          }
+
+          // Call compile to ensure that the returned Module is ready to run.
+          loweredModule.run_method("__compile__");
```

Comment:
I think we should call `__setstate__` here directly, so that we test the exact path the `load()` will take.